code/go/0chain.net/smartcontract/dbs/event/validator.go

```diff
@@ -109,10 +111,34 @@ func (edb *EventDb) updateValidators(validators []Validator) error {
 		"delegate_wallet", "num_delegates",
 		"service_charge",
 	}
-	return edb.Store.Get().Clauses(clause.OnConflict{
-		Columns:   []clause.Column{{Name: "id"}},
-		DoUpdates: clause.AssignmentColumns(updateFields),
-	}).Create(&validators).Error
+
+	// Create column-based listing of the given data
+	columns, err := util.Columnize(validators)
+	if err != nil {
+		return err
+	}
+
+	// Create the updater
+	ids, ok := columns["id"]
+	if !ok {
+		return common.NewError("update_validators", "no id field provided in event Data")
+	}
+	updater := CreateBuilder("validators", "id", ids)
+
+	// Bind the required fields for update to the updater
+	for _, fieldKey := range updateFields {
+		if fieldKey == "id" {
+			continue
+		}
+
+		fieldList, ok := columns[fieldKey]
+		if !ok {
+			return common.NewErrorf("update_validators", "required field %v for update is not found in provided data", fieldKey)
+		}
+		updater = updater.AddUpdate(fieldKey, fieldList)
```

Comment:
same as for allocation

Reply:
That would be easier for me, but, as I mentioned above, I just didn't like to add any fields to the update that wasn't there before.